GetStream/stream-video-android
# Feature/xml screenshare

### 🎯 Goal

Add ScreenShareView to XML sdk to preview screenshares.

### 🛠 Implementation details

Added `ScreenShareView`. Refactored `CallParticipantsView` to use `CallParticipantsGridView` when outside screen share and `CallParticipantsListView` and `ScreenShareView` when inside.

### 🎨 UI Changes

| Screenshare |
| --- |
| ![Screenshot_1677143158](https://user-images.githubusercontent.com/38032787/220873420-4c91eac5-0248-4e27-8046-8d0e911b62cd.png) |

### 🧪 Testing

<details>

<summary>XML patch</summary>

```
Index: dogfooding/src/main/java/io/getstream/video/android/dogfooding/DogfoodingApp.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dogfooding/src/main/java/io/getstream/video/android/dogfooding/DogfoodingApp.kt b/dogfooding/src/main/java/io/getstream/video/android/dogfooding/DogfoodingApp.kt
--- a/dogfooding/src/main/java/io/getstream/video/android/dogfooding/DogfoodingApp.kt	(revision 04560f56ea17a45073ec8da6118a2fcfbac4d88f)
+++ b/dogfooding/src/main/java/io/getstream/video/android/dogfooding/DogfoodingApp.kt	(date 1677145298633)
@@ -68,8 +68,8 @@
             pushDeviceGenerators = listOf(FirebasePushDeviceGenerator()),
             androidInputs = setOf(
                 CallServiceInput.from(CallService::class),
-                // CallActivityInput.from(XmlCallActivity::class),
-                CallActivityInput.from(CallActivity::class),
+                CallActivityInput.from(XmlCallActivity::class),
+                // CallActivityInput.from(CallActivity::class),
             )
         ).build().also {
             video = it

```

</details>

1. Checkout this branch and apply the patch above
2. Run the dogfooding app on at least one phone/emu and on the web client
3. Start screen share on the web client and make sure screen share content is visible
4. Stop screen share and make sure layout updates
5. Try entering the call while screen share is active

<details>

<summary>Style test</summary>

```
Index: stream-video-android-xml/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-video-android-xml/src/main/res/values/styles.xml b/stream-video-android-xml/src/main/res/values/styles.xml
--- a/stream-video-android-xml/src/main/res/values/styles.xml	(revision 04560f56ea17a45073ec8da6118a2fcfbac4d88f)
+++ b/stream-video-android-xml/src/main/res/values/styles.xml	(date 1677145558422)
@@ -108,6 +108,8 @@
     </style>
 
     <style name="Stream.ScreenShare" parent="Stream">
+        <item name="streamScreenShareControlsBackgroundTint">#FFED00</item>
+        <item name="streamScreenShareControlsIconTint">#FF0000</item>
     </style>
 
 


```

</details>

1. Apply the patch above keeping the first one
2. Start dogfooding, enter a call and start screen share
3. The screen share button style should be applied

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![giphy (2)](https://user-images.githubusercontent.com/38032787/220873912-637e55f1-4bf2-40a2-9391-9f45bba650c9.gif)
